src/sentry/api/endpoints/organization_events_trends.py

```diff
@@ -152,9 +153,20 @@ def get(self, request, organization):
             return Response([])
 
         with sentry_sdk.start_span(op="discover.endpoint", description="trend_dates"):
-            middle = params["start"] + timedelta(
-                seconds=(params["end"] - params["start"]).total_seconds() * 0.5
-            )
+            middle_date = request.GET.get("middle")
+            if middle_date:
+                try:
+                    middle = parse_datetime_string(middle_date)
+                except InvalidQuery:
+                    raise ParseError(detail="{} is not a valid date format".format(middle_date))
+                if middle <= params["start"] or middle >= params["end"]:
+                    raise ParseError(
+                        detail="The middle date should be within the duration of the query"
+                    )
+            else:
+                middle = params["start"] + timedelta(
+                    seconds=(params["end"] - params["start"]).total_seconds() * 0.5
```

Comment:
Do you need to round here to avoid partial seconds?

Reply:
No I don't think so, timedelta will handle partial seconds for us

double checking this, looks like timedelta only stores days seconds and microseconds and will round to the nearest microsecond, which is fine for how it is being used here